src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -1177,6 +1180,78 @@ void CompilationMemoryStatisticDCmd::execute(DCmdSource source, TRAPS) {
   CompilationMemoryStatistic::print_all_by_size(output(), human_readable, minsize);
 }
 
+VMDebugDCmd::VMDebugDCmd(outputStream* output, bool heap) :
+                                     DCmdWithParser(output, heap),
+  _subcommand("subcommand", "", "STRING", true, nullptr),
+  _arg1("arg1", "", "STRING", false, nullptr),
+  _arg2("arg2", "", "STRING", false, nullptr),
+  _arg3("arg3", "", "STRING", false, nullptr),
+  _verbose("-verbose", "", "BOOLEAN", false, "false") {
+
+  _dcmdparser.add_dcmd_argument(&_subcommand);
+  _dcmdparser.add_dcmd_argument(&_arg1);
+  _dcmdparser.add_dcmd_argument(&_arg2);
+  _dcmdparser.add_dcmd_argument(&_arg3);
+  _dcmdparser.add_dcmd_option(&_verbose);
+}
+
+void VMDebugDCmd::find() {
+  if (!_arg1.has_value()) {
+    output()->print_cr("missing argument");
```

Comment:
I'm thinking if it would be useful to tell what arguments are expected? This is for all cases where the `"missing argument"` message is returned.

Reply:
Thanks Serguei, have filled in explicit usage instructions when the commands are missing an argument.